scripts/sagas/collection.js

```diff
@@ -55,19 +55,16 @@ export function* listRecords(getState, action) {
 
 export function* createRecord(getState, action) {
   const {session} = getState();
-  const {bid, cid, record} = action;
+  const {bid, cid, record: rawRecord} = action;
   try {
     const coll = getCollection(bid, cid);
     yield put(actions.collectionBusy(true));
-    if (shouldProcessAttachment(session.serverInfo, record)) {
-      const rid = yield call(uuid);
-      const formData = yield call(createFormData, record);
-      yield call(requestAttachment, bid, cid, rid, {
-        method: "post",
-        body: formData
-      });
+    if (shouldProcessAttachment(session.serverInfo, rawRecord)) {
+      const id = yield call(uuid);
+      const {__attachment__: dataURL, ...record} = rawRecord;
```

Comment:
I can probably update the action payload to separate between record data and attachment to ease reading the code.